### ISSUE TITLE
Zones (synchronized from Firebase) selector

### ISSUE THREAD

**Issue by HormCodes** [GDGJihlava/parkovani-v-jihlave#2]
I have got an idea, that we can implement zone selector instead of EditText. I think that zones could be updated from Firebase database although zones will not be updated very often.  What do you say?

**Comment by leosprikryl** [comment]
I agree. It was my original idea as well.

**Comment by HormCodes** [comment]
@leosprikryl can I get acces to Firebase Project?

**Comment by leosprikryl** [comment]
@Horm access granted

**Comment by HormCodes** [comment]
I suggested a structure into Firebase Database. The structure is based on UI which will be as two selectors (zone code or parking name). One selector automatically changes the second. Can you please review the structure?

**Comment by leosprikryl** [comment]
I wouldn't use the two selectors in the UI in the first place. I would use just one - location. User doesn't care (and might even not know) what zone it is. He just knows the location where he left his car. 
It would be nice to have GPS coordinates as well, so we can detect the location automatically. Then it would be really user-friendly as user wouldn't have to know anything.

If you agree with the suggestion, you should change the data structure accordingly. The relation between locations and zones should lead from locations to zones, not in the opposite direction as it is now. 
And it would be nice to include some fields for GPS - either single point or maybe top left and bottom right corner. This can of course be added later, if you don't want to do it right now. It is not necessary for the first version.

**Comment by HormCodes** [comment]
I agree and I updated the structure. Can you please review it again? 

Automatic selecting by location is a great feature but it's a huge step and I think it can wait for now.

**Comment by leosprikryl** [comment]
The new data structure looks good. 

I just don't like the word "parkings". It doesn't sound right to me. See https://forum.wordreference.com/threads/parking.1312264/ for a discussion about that. But it is a really minor detail.

**Comment by HormCodes** [comment]
I implemented the selector. I also implemented confirm dialog which shows information about a ticket. In future, I want to add this information to Activity directly. 

@leosprikryl, can you please look at getting data from Firebase in [ParkingLotSelector class](https://github.com/GDGJihlava/parkovani-v-jihlave/blob/zone-selector/app/src/main/java/cz/gdgjihlava/parkovani/parkovani_v_jihlave/sms/ParkingLotSelector.java)? I need an advice how to refactor these nested code structures. 

Next notes how to improve my code or selector functionality are welcomed.

**Comment by leosprikryl** [comment]
You don't have to read the data from Firebase key by key. Firebase API can map structures to Java objects/POJOs.
Just create a POJO with default constructor, getters and setters and then use it as argument of the `.getValue ` method. That would allow you to get rid of the key names constants and it would simplify your code.

```java
public class ParkingLot {

    private String name;
    private String zoneId;

   //...
   //getters and setters
   //...
}

ParkingLot parkingLot = parkingLotSnapshot.getValue(ParkingLot.class)
```
And I think you should split the ParkingLotSelector to several smaller methods - something like `readParkingLot` and `readZone`.
